tests/testthat/test-OTHER-create-track-hub.R

```diff
@@ -0,0 +1,154 @@
+    bw.dt <- readBigWig(bigWig.file, "chr10", 0, 128616069)
+    out.dt <- data.table(chrom="chr10", bw.dt)
+    demo.bedGraph <- sub("bigWig", "bedGraph", demo.bigWig)
+    fwrite(out.dt, demo.bedGraph, sep="\t", col.names=FALSE)
+    cmd <- paste("bedGraphToBigWig", normalizePath(demo.bedGraph), normalizePath(chrom.sizes.file), normalizePath(demo.bigWig))
+    system.or.stop(cmd)
+    unlink(demo.bedGraph)
+  }
+}
+
+
+sample.dir <- dirname(demo.bigWig)
+problem.dir <- file.path(sample.dir, "problems", "chr10:18024675-38818835")
+coverage.bedGraph <- file.path(problem.dir, "coverage.bedGraph")
+unlink(coverage.bedGraph)
+
+for(set.dir in c(non.integer.dir, demo.dir)){
+  labels.file <- file.path(set.dir, "labels", "some_labels.txt")
+  dir.create(dirname(labels.file), showWarnings=FALSE, recursive=TRUE)
+  writeLines(label.txt, labels.file)
+  problems.bed <- file.path(set.dir, "problems.bed")
+  unlink(problems.bed)
+  cat("chr10	18024675	38818835", file=problems.bed)
+}
+
+unlink(non.integer.dir, recursive=TRUE, force=TRUE)
+
+## Set time limit.
+(sample.dir.vec <- Sys.glob(file.path(
+  demo.dir, "samples", "*", "*")))
+prob.dir.vec <- file.path(
+  sample.dir.vec, "problems", "chr10:18024675-38818835")
+limit.dt <- data.table(minutes=5)
+for(prob.dir in prob.dir.vec){
+  dir.create(prob.dir, showWarnings=FALSE, recursive=TRUE)
+  limit.file <- file.path(prob.dir, "target.minutes")
+  fwrite(limit.dt, limit.file, col.names=FALSE)
+}
+
+# Step 0
+convert_labels(demo.dir)
+# Step 1
+problem.dir.glob <- file.path(demo.dir, "samples", "*", "*", "*", "*")
+problem.dir.vec <- Sys.glob(problem.dir.glob)
+for(problem.dir in problem.dir.vec){
+  problem.target(problem.dir)
+}
+# Step 2
+problem.train(demo.dir)
+# Step 3
+prob.dir <- file.path(demo.dir, "problems", "chr10:18024675-38818835")
+problem.pred.cluster.targets(prob.dir)
+# Step 4
+problem.joint.train(demo.dir)
+# Step 5
+job <- file.path(demo.dir, "jobs", "1")
+problem.joint.predict.job(job)
+# Step 6
+plot_all(demo.dir)
+
+url <- paste0("http://CHANGE.THIS/~URL/", basename(demo.dir))
+email <- "toby.hocking@r-project.org"
+genome <- "hg19"
+create_track_hub(demo.dir, url, genome, email)
+
+test_that("hub.txt file is created", {
+  expect_true(file.exists(hub.txt))
+})
+
+test_that("hub.txt file has correct email", {
+  hub.vec <- readLines(hub.txt)
+  index <- length(hub.vec)
+  hub.txt.email <- substring(hub.vec[index], 7)
+  expect_equal(hub.txt.email, email)
+})
+
+test_that("trackDb.txt file has the correct link to the bigwig files", {
+  trackDb.vec <- readLines(trackDb.txt)
+  kidney.coverage.line <- grep("samples/kidney/MS002201/coverage.bigWig", trackDb.vec)
+  kidney.jointpeak.line <- grep("samples/kidney/MS002201/joint_peaks.bigWig", trackDb.vec)
+  bcell.coverage.line <- grep("samples/bcell/MS010302/coverage.bigWig", trackDb.vec)
+  bcell.jointpeak.line <- grep("samples/bcell/MS010302/joint_peaks.bigWig", trackDb.vec)
+  split.index <- 14
```

Comment:
can you remove this constant (14) please?

Reply:
Should I just remove split.index constant and instead use its value directly in function call[substring]? Or Should I rewrite the code using strsplit ?

Reply:
please remove all numeric constants from the test if possible. use names/text that clarify your intent instead. e.g. `after.space <- sub(".* ", "", trackDb.vec)`